ui.go

```diff
@@ -5,24 +5,95 @@ import (
 	"log"
 	"net/http"
 	"os"
+	"strconv"
+	"github.com/gorilla/websocket"
 )
 
-func parseBenchmarkingForm(w http.ResponseWriter, r *http.Request) {
+var connections map[*websocket.Conn]bool
+var statChan chan GlobalStatRecord
 
+//For later. Will be used to parse config inputs.
+func parseBenchmarkingForm(w http.ResponseWriter, r *http.Request) {
 	decoder := json.NewDecoder(r.Body)
 	var benchmarkMap map[string]interface{}
 	err := decoder.Decode(&benchmarkMap)
 	if err != nil {
 		panic(err)
 	}
-	log.Println(benchmarkMap["benchmarkInput"])
+	log.Println(benchmarkMap["configName"])
+}
+
+func closeConnectionListener(conn *websocket.Conn, quit chan *websocket.Conn){
+	for {
+		_, msg, err := conn.ReadMessage()
+		if err != nil {
+			delete(connections, conn)
+			conn.Close()
+			return
+		}
+		if(string(msg) == "quit"){
+			quit <- conn
+		}
+	}
+}
+
+func socketHandler(w http.ResponseWriter, r *http.Request) {
+	conn, err := websocket.Upgrade(w, r, nil, 1024, 1024)
+	if _, ok := err.(websocket.HandshakeError); ok {
+		http.Error(w, "Not a websocket handshake", 400)
+		return
+	} else if err != nil {
+		log.Println(err)
+		return
+	}
+
+	log.Println("Succesfully upgraded connection")
+	connections[conn] = true
+
+
+	quit := make(chan *websocket.Conn)
+	//go func() {
+	    for {
+	       select {
+	       	case astat := <- globalStatSink:
```

Comment:
After user presses Start button I create a web socket [connection](https://github.com/tugceakin/bouncer/blob/master/public/js/services/benchmarking.js#L60) and [start listening](https://github.com/tugceakin/bouncer/blob/master/public/js/services/benchmarking.js#L74). 

Each time globalStatSink is updated [I send the data to the UI](https://github.com/tugceakin/bouncer/blob/master/ui.go#L77):